dhis-2/dhis-test-integration/src/test/java/org/hisp/dhis/maintenance/MaintenanceServiceTest.java

```diff
@@ -260,7 +262,8 @@ void testDeleteSoftDeletedEnrollmentWithAProgramMessage()
     manager.save(enrollment);
     programMessageService.saveProgramMessage(message);
     assertNotNull(manager.get(Enrollment.class, enrollment.getUid()));
-    apiEnrollmentService.deleteEnrollment(enrollment);
+
+    trackerObjectDeletionService.deleteEnrollments(List.of(enrollment.getUid()));
```

Comment:
is this what we want @enricocolasante ? should we not simply use the `manager` here? who cares about the `status`. I would rather not add a dependency from here to the deletionService.

Reply:
The `status` will be correctly updated using the `manager` as in the end the correct store will be used.
The difference is that using `TrackerObjectDeletionService` will delete enrollments and all linked events and relationships. 
Anyway I think it is better to use the `manager` and manually delete any linked entity.